libraries/DataFlash/DataFlash.cpp

```diff
@@ -681,12 +681,45 @@ void DataFlash_Class::Log_WriteV(const char *name, const char *labels, const cha
 }
 
 
+void DataFlash_Class::assert_same_fmt_for_name(const DataFlash_Class::log_write_fmt *f,
+                                               const char *name,
+                                               const char *labels,
+                                               const char *units,
+                                               const char *mults,
+                                               const char *fmt) const
+{
+    if (!streq(f->name, name)) {
+        // why exactly were we called?!
+        AP_HAL::panic("format names differ");
+    }
+    if (!streq(f->labels, labels)) {
+        AP_HAL::panic("format labels differ");
+    }
+    if ((f->units != nullptr && units == nullptr) ||
+        (f->units == nullptr && units != nullptr) ||
+        (units !=nullptr && !streq(f->units, units))) {
+        AP_HAL::panic("format units differ");
+    }
+    if ((f->mults != nullptr && mults == nullptr) ||
+        (f->mults == nullptr && mults != nullptr) ||
+        (mults != nullptr && !streq(f->mults, mults))) {
+        AP_HAL::panic("format mults differ");
+    }
+    if (!streq(f->fmt, fmt)) {
+        AP_HAL::panic("format fmt differ");
```

Comment:
I really tried to get up the enthusiasm to fill these out, I really did.  These codepaths will almost likely *never* be crossed - you have to try to define a message which already exists via `Log_Write`, after all.  This adds sanity checking which wasn't there before and will start the user on the right path - not really sure we need to do anything more than that.

I'll have a look at it, anyway.

Reply:
Yes, I understand we shouldn't hit this often, if ever, but having a bit more info, like the format name is useful. Same for printing all messages, it helps the developer fix all problems at once instead of fixing one and then finding out it still isn't OK (I guess that is the reason we do it with the validation).